api/rs/slint/tests/simple_macro.rs

```diff
@@ -15,3 +15,15 @@ fn empty_stuff() {
     slint!(export struct Hei { abcd: bool });
     slint!(export global G { });
 }
+
+#[cfg(all(feature = "serde", feature = "serde-json"))]
+use slint::SharedVector;
+
+#[cfg(all(feature = "serde", feature = "serde-json"))]
+#[test]
+fn test_serialize_deserialize() {
+    let v = SharedVector::from([1, 2, 3]);
+    let serialized = serde_json::to_string(&v).unwrap();
+    let deserialized: SharedVector<i32> = serde_json::from_str(&serialized).unwrap();
+    assert_eq!(v, deserialized);
```

Comment:
that's great to test the serialization of vector.
We should also test the serialization of struct declared in a slint! macro.

Reply:
how can we use struct defined in sline! ?

Reply:
Do you think can you provide it ? I suppose need expertise like you in slint and its syntax

Reply:
can we do something like this ?
```rust
#[test]
fn test_serialize_deserialize_person() {
    i_slint_backend_testing::init();
    let mut compiler = slint_interpreter::ComponentCompiler::default();
    compiler.set_style("fluent".into());
    let comp_def = spin_on::spin_on(
        compiler.build_from_source(
            r#"
            export struct TestStruct {
                foo: int,
                bar: string,
            }
    
            export component Test {
                in property <TestStruct> data: [
                    {foo: 1, bar: "test"},
                    {foo: 2, bar: "example"},
                ];
            }
            "#
            .into(),
            "".into(),
        ),
    )
    .unwrap();

    let props = comp_def.properties().collect::<Vec<(_, _)>>();
}
```
for get struct ?